### DISCUSSION TITLE
I call an await  api in SaveHandler but it is not wait response

### DISCUSSION THREAD

**Discussion by wow-prime** [serenity-is/Serenity#7298] [Help] [answered]
I call an await  api in SaveHandler but it is not wait response.
My Code is below:

protected async override void ExecuteSave()
{
    base.ExecuteSave();
    string strResponseBody = "";
    if (Row.StsAdpy == "10")
    {
            using (HttpClient client = new HttpClient())
            {
                Dictionary<string, string> formDataDictionary = new Dictionary<string, string>()
                {
                    {"sNumAdpy", Row.NumAdpy }

                };

                client.Timeout = TimeSpan.FromSeconds(300);
           
                var formData = new FormUrlEncodedContent(formDataDictionary);

                using (var response = await client.PostAsync(_config.GetValue<string>("Sbpm") + "FI/ApiPrePaidIns", formData))
                {
                    strResponseBody = await response.Content.ReadAsStringAsync();
                  

                    Row.StsAdpy = "00";

                    if (strResponseBody == "")
                        throw new ValidationError("驗收失敗：" + strResponseBody);
                    throw new ValidationError("驗收失敗：" + strResponseBody);

                }

            }
    }
}

How can I modify it so that I can continue the next action after getting a reply?

**Comment by VictorTomaili** [comment, marked as answer]
handlers are not support async await. They are chain of methods should run one by one and we don't have async overrides.
